client.go

```diff
@@ -182,6 +223,19 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		req.Method, req.URL, c.RetryMax+1)
 }
 
+// Try to read the response body so we can reuse this connection.
+func (c *Client) drainBody(body io.ReadCloser) {
+	defer body.Close()
+	_, err := io.Copy(ioutil.Discard, &io.LimitedReader{R: body, N: respReadLimit})
+	if err != nil {
+		// don't log this if the body was closed by the ResponseLogHook or CheckRetry
+		if strings.Contains(err.Error(), "read on closed response body") {
```

Comment:
Yeah, I didn't like this either. I think it was required for some tests on an earlier iteration, but is no longer needed. I'll just pull this out and make a note in the docs.